elasticsearch.md

```diff
@@ -1719,64 +1718,134 @@ readonlyrest:
 
 ```bash
 $ export LDAP_PASSWORD=S3cr3tP4ss
 $ cat readonlyrest.yml
 ```
 
-```text
-....
+```yaml
 ldaps:
-
-    - name: ldap1
-      host: "ldap1.example.com"
-      port: 389                                                     
-      ssl_enabled: false                                            
-      ssl_trust_all_certs: true                                     
-      bind_dn: "cn=admin,dc=example,dc=com"                         
-      bind_password: "${LDAP_PASSWORD}"                                     
-      search_user_base_DN: "ou=People,dc=example,dc=com"
-....
+  - name: ldap1
+    host: "ldap1.example.com"
+    port: 389                                                     
+    ssl_enabled: false                                            
+    ssl_trust_all_certs: true                                     
+    bind_dn: "cn=admin,dc=example,dc=com"                         
+    bind_password: "${env:LDAP_PASSWORD}"                                     
+    search_user_base_DN: "ou=People,dc=example,dc=com"
 ```
 
 And ReadonlyREST ES will load "S3cr3tP4ss" as `bind_password`.
 
 ### Dynamic variables
 
-One of the neatest feature in ReadonlyREST is that you can use dynamic variables inside most rules values. The variables you can currently replace into rules values are these:
+One of the neatest features in ReadonlyREST is that you can use dynamic variables inside most values of the following rules: `indices`, `users`, `groups`, `groups_and`, `fields`, `filter`,  `repositories`, `snapshots`, `response_fields`, `uri_re`, `x_forwarded_for`, `hosts_local`, `hosts`, `kibana_index`, `kibana_template_index`. The variables are related to different contexts:
+* `acl` - the context of data collected in authentication and authorization rules of the current block:
+    * `@{acl:user}` gets replaced with the username of the successfully authenticated user. Using this variable is allowed only in blocks where one of the rules is an authentication rule of course it must be rule different from the one containing the given variable.
+    * `@{acl:current_group}` gets replaced with user's current group. Usually resolved by authorization rule defined in a block, but value can be also retrieved by means of kibana plugin. This variable doesn't specify usage requirements.
+    * `@{acl:available_groups}` gets replaced with available groups found in the authorization rule (bacause group strings will be surrounded with double quotes and joined with a comma)
+* `header` - the context of ES HTTP request headers
+    * `@{header:<header_name>}` gets replaced with the value of the HTTP header with name `<header_name>` included in the incoming request \(useful when reverse proxies handle authentication\)
+* `jwt` - the context of JWT passed in the Authorization header 
+    * `@{jwt:<json_path>}` get replaced with value (or values) found in JWT under the given JSON path
```

Comment:
"gets replaced with the value (or values) found in the JWT claim under the given JSON path."